src/main/java/com/hivemq/mqtt/message/publish/PUBLISH.java

```diff
@@ -408,7 +409,9 @@ public int getEstimatedSizeInMemory() {
         size += ObjectMemoryEstimation.stringSize(contentType);
 
         size += 24; //User Properties Overhead
-        for (final MqttUserProperty userProperty : getUserProperties().asList()) {
+        final ImmutableList<MqttUserProperty> userProperties = getUserProperties().asList();
+        for (int i = 0; i < userProperties.size(); i++) {
```

Comment:
Why change the for each loop?  The index isn't used.

Reply:
The for each was visible in the sampler because the iterator is more expensive.
ImmutableLists are random access, so access by index is very cheap.

Reply:
Unexpected but ok.